ark/schema/shared/errors.ts

```diff
@@ -279,6 +287,24 @@ export class ArkErrors
 	}
 }
 
+export class TraversalError extends Error {
+	name = "TraversalError"
+	arkErrors!: ArkErrors
+
+	constructor(errors: ArkErrors) {
+		if (errors.length === 1) super(errors.summary)
+		else super("\n" + errors.map(error => `  • ${indent(error)}`).join("\n"))
+
+		Object.defineProperty(this, "arkErrors", {
```

Comment:
I guess my main concern is that this is more breaking if people were catching an AggregateError and accessing the errors prop.

If there's no way to use that name without having it serialized though not sure what the alternative is

Reply:
If they were doing this without `instanceof AggregateError` I think their code was probably pretty brittle anyways. I'll take another look at if it's possible to add `errors` as a prop without logging though.

Reply:
I tried:
- a non-enumerable `errors` property.
- a `get`/`set` pair for `errors`.
- a `get`-only item for `errors`.
- a non-configurable, non-enumerable, non-writable `errors` property
- looking through `Symbol` in case I could override `in` or something.

Unfortunately I did not figure out a way to have `errors` not be serialized in the message. I think it's safe to say most users of ArkType write in TypeScript and it's unlikely anyone's writing `try { type("number")("foo") } catch (err) { err.errors }` for the kick of it.